server/SCHEMAS/mongodb.js

```diff
@@ -11,33 +11,131 @@ db.on("error", console.error.bind(console, "MongoDB connection error:"))
 const Schema = mongoose.Schema
 
 const AnswerSchema = new Schema({
-  asnwer_id: Number,
-  body: String,
-  date: Date,
+  answer_id: {
+    type: Number,
+    unique: true,
+    sparse: true
+  },
+  answer_body: String,
+  answer_date: Date,
   answerer_name: String,
-  helpfulness: Number,
-  photos: Array,
-  question_ref: { type: Schema.Types.ObjectId, ref: "Question" }
+  answerer_email: String,
+  answer_reported: Boolean,
+  answer_helpfulness: Number,
+  answer_photos: [String]
 })
 
-const QuestionSchema = new Schema({
-  product_id: String,
-  question_id: Number,
+const QandASchema = new Schema({
+  question_id: {
+    type: Number,
+    index: true,
+    unique: true
+  },
+  product_id: Number,
   question_body: String,
   question_date: Date,
   asker_name: String,
+  asker_email: String,
+  question_reported: Boolean,
   question_helpfulness: Number,
-  reported: Boolean,
-  answers: [{ type: Schema.Types.ObjectId, ref: "Answer" }]
+  answers: [AnswerSchema]
 })
 
+let insert = async (data) => {
+  await QandA.create(data)
+    .then(() => {})
+    .catch((err) => console.log('err in insert', err.message))
+}
+
+let answerInsert = async (question_id, data) => {
+  // query the db for the question and needed info
+  let query = { question_id }
+  let question = await QandA.find(query)
+  let id = question[0]._id
+  let answers = question[0].answers
+
+  // setup the new answer data
+  let answer = new Answer({
+    answer_id: data.id,
+    answer_body: data.body,
+    answer_date: data.date_written,
+    answerer_name: data.answerer_name,
+    answerer_email: data.answerer_email,
+    answer_reported: data.reported,
+    answer_helpfulness: data.helpful,
+    answer_photos: []
+  })
+
+  // if not present, insert the answer to the array from the question
+  let isPresent = false
```

Comment:
We talked about this part and concluded that there is likely a better way to do this. Perhaps using mongodb's $set? Maybe for a future tech debt ticket!